### PR TITLE
Add dev container configuration for using with Codespaces

### PR DESCRIPTION
**Pull request by pepone**
This PR introduces the necessary dev container configuration for development in Codespaces. It includes a Dockerfile that sets up the .NET and Rust development environments required for building IceRPC for C#.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "C# (.NET)",
+    // Use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+    "build": {
+        "dockerfile": "Dockerfile"
+    },
+    // Configure tool-specific properties
+    "customization": {
+        "vscode": {
+            "extensions": [
```
Comment (externl) [inline .devcontainer/devcontainer.json:10]: Missing the most important extension: https://marketplace.visualstudio.com/items?itemName=ZeroCInc.slice :)

File: .vscode/extensions.json [hunk 2]
```diff
@@ -1,8 +1,9 @@
 {
     "recommendations": [
         "EditorConfig.EditorConfig",
```
Comment (externl) [inline .vscode/extensions.json:3]: Ditto

**Review by externl — APPROVED** [review]
Looks good!

**Review by ReeceHumphreys — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (pepone) [reply .devcontainer/devcontainer.json:10]: Added

**Pull request merged**